### ISSUE TITLE
Where to "install" appimagekit?

### ISSUE THREAD

**Issue by nubnubbud** [Ultimaker/cura-build#248]
the build guide says to install appimagekit, but to the best of my knowledge, I can't tell where. I've put it path, on my desktop, in various files in cura source, but nothing works, I just get:

```
CMake Error at /usr/share/cmake-3.16/Modules/FindPackageHandleStandardArgs.cmake:146 (message):
  Could NOT find AppImageKit (missing: APPIMAGEKIT_APPIMAGETOOL_EXECUTABLE
  APPIMAGEKIT_APPRUN_EXECUTABLE)
Call Stack (most recent call first):
  /usr/share/cmake-3.16/Modules/FindPackageHandleStandardArgs.cmake:393 (_FPHSA_FAILURE_MESSAGE)
  cmake/FindAppImageKit.cmake:5 (find_package_handle_standard_args)
  packaging/python3.7/linux.cmake:2 (find_package)
  CMakeLists.txt:124 (include)


-- Configuring incomplete, errors occurred!
```
every time, and as far as I can tell there's no right way to do this. The only other information I have is from a previous person who had the same error, and the only real response was to look at a docker file with a smiley, and nothing there shows any distinct directories or directions about appimagekit, so that's not helping me much, and the docker file seems only for centOS, or at the very least I have no clue what that page is on about. Docker is confusing compared to cloning a git repo and siccing Cmake on it.

**Comment by nallath** [comment]
Did you already try just setting APPIMAGEKIT_APPIMAGETOOL_EXECUTABLE to be the path of the appimage-builder executable?

**Comment by nubnubbud** [comment]
~~well, I put the .appimage in path and made it executable... but I'm not sure what to make of an executable that isn't an executable, yet cmake will look for like a library. How do I set that to the path? I might not be doing it right~~

~~but the thing is... I can't find a suitable release. there... doesn't seem to be a pure executable. they're all appimages, appruns, and... unmarked.~~

alright, it was super confusing so I was able to find the cmake file and open it up and...
no appimagekit releases work without being edited. Apprun has no upfront download page, and only has releases through its github, which are all marked with the OS architecture it targets. basically, cmake was looking for appimagetool.appimage and apprun, but it was getting appimagetool-x86_64.appimage and apprun-x86_64

so unless I looked at the findappimagekit.cmake or knew about the intricacies of cmake (honestly I'm impressed anyone can- it's super cryptic), it wouldn't have worked, even though I put it in path and linked to it 3 different correct ways. man, I'd like to take a scalpel to the build instructions and make them a bit more straightforward XD.

**TL;DR: I was setting it to the right path and getting the right files, but the cmake file is looking for filenames appimagekit is not released with.**

**Comment by Ghostkeeper** [comment]
> no appimagekit releases work without being edited. Apprun has no upfront download page, and only has releases through its github, which are all marked with the OS architecture it targets. basically, cmake was looking for appimagetool.appimage and apprun, but it was getting appimagetool-x86_64.appimage and apprun-x86_64

For me on Ubuntu, installing through `apt install libappimage0` worked.

> now I'm getting this- which I don't think should be happening-is it true only one version of protobuf will work?

No, that's not true. The official build of Cura still uses [Protobuf 3.0.2](https://github.com/Ultimaker/cura-build-environment/blob/a4d1ec909c4ac65220f0a7aa915e082bb277ee36/projects/protobuf.cmake#L1) (as installed through cura-build-environment). The cura-build repository doesn't specify any Protobuf version. I don't know how it got to version 3.9.2.0 on your system other than that being the Protobuf version you actually installed. But apparently it didn't install well, then.
